corgidrp/calibrate_kgain.py

```diff
@@ -0,0 +1,1099 @@
+import os
+from pathlib import Path
+import numpy as np
+import warnings
+from scipy.optimize import curve_fit
+
+import corgidrp.data as data
+from corgidrp.data import Image
+from corgidrp.mocks import create_default_headers
+
+class CalKgainException(Exception):
+    """Exception class for calibrate_kgain."""
+
+def calibrate_kgain(stack_arr, stack_arr2, emgain, min_val, max_val, 
```

Comment:
Oh, and one other thing:  An error for the k gain could be obtained and put into an `err` of the `KGain` class instantiation that makes the calibration product.  That would require more than a mere port of this code, but I don't think it wouldn't be too hard to implement.  I didn't know how precise we wanted to be with that.  Perhaps that could be made a GitHub issue as a possible enhancement?

Reply:
> Yes, I think it would be good to have the output be a KGain class (ideally with error, but I don't think that's a hard requirement at this point. If it's a pain to put in here we can make it a feature request for late.

@maxwellmb and @kjl0025 : the return has been a KGain class [link](https://github.com/roman-corgi/corgidrp/blob/e19f15ce1e537df724f133446591c964fa432715/corgidrp/calibrate_kgain.py#L689). Can you explain what's missing?

Reply:
> For the input, it should be just a single Dataset. Internal to this function we can do the sorting as need be. The EMGain should be in the header of Dataset objects, so I don't think we need `emgain` as an input parameter. `min_val` and `max_val` should be set as keyword arguments.

calibrate_kgain (and calibrate_nonlin) have stacks (of stacks) of arrays [docstring](https://github.com/roman-corgi/corgidrp/blob/e19f15ce1e537df724f133446591c964fa432715/corgidrp/calibrate_kgain.py#L16) Initially, @kjl0025 and I thought to build stacks (of stcks) of Datasets. @maxwellmb: can you explain what you mean by having a single Dataset instead?

Reply:
I think just get rid of the other outputs and put them into headers/extensions should be fine.

Reply:
@maxwellmb header/extensions of which variable?

Reply:
Also, if the only output from calibrate_kgain and calibrate_nonlin is either a KGain or NonLinearity object, respectively: how are these keywords/extensions output? The input variables are copied as per corgidrp best practices. Therefore, changes inside the code will not be accessible after the function is run, unless they are returned.

Reply:
I just meant that you should put those outputs into the KGain object header, and if need be, feel free to add an extension for arrays (like the PTC).

Reply:
I just meant that you should put those outputs into the KGain object header, and where appropriate (e.g. for an array of values) add an extension HDU for arrays, like the PTC (including useful headerwords about the content of the extension, and a name, per the developer guidelines).

Reply:
Working on the *input* datasets (stack_arr and stack_arr2)

Reply:
Done. Kgain value and error are returned as part of a single object